components/canvas/canvas_paint_thread.rs

```diff
@@ -118,9 +118,10 @@ impl<'a> CanvasPaintThread<'a> {
     /// communicate with it.
     pub fn start(size: Size2D<i32>,
                  webrender_api_sender: webrender_api::RenderApiSender,
-                 antialias: bool)
+                 antialias: bool,
+                 canvas_channel: (IpcSender<CanvasMsg>, IpcReceiver<CanvasMsg>))
                  -> IpcSender<CanvasMsg> {
```

Comment:
Since all this method does is return the sender, we can make this code more efficient by only accepting the IpcReceiver argument and not returning anything.